src/vulkan-renderer/wrapper/texture.cpp

```diff
@@ -39,24 +39,38 @@ Texture::Texture(const wrapper::Device &device, const VkPhysicalDevice graphics_
 
     // Load the texture file using stb_image library.
     // Force stb_image to load an alpha channel as well.
-    stbi_uc *texture_data =
-        stbi_load(file_name.c_str(), &m_texture_width, &m_texture_height, &m_texture_channels, STBI_rgb_alpha);
+    stbi_uc* texture_data =
+        stbi_load(file_name.c_str(), &m_texture_width, &m_texture_height, nullptr, STBI_rgb_alpha);
```

Comment:
I think this change is interesting.

To me, this texture class is a wrapper around the data for the graphics card.  Or, said differently, the spirit of the class is to represent GPU texture data and not file system image data.

So, it is weird to me to be storing the file system image channel definition when we are altering it to add an additional channel for the GPU.  If I was debugging and looking at texture instances and m_texture_channels was 3, I would expect 3 channels on the GPU texture data - and that was not currently the case.

So, anyways, I updated that in these changes.  I think that is more correct.

Reply:
I agree, I think it would be nicer if we had some kind of separate CPU and GPU texture class, especially since both textures are in memory on different devices, so enclosing both in one wrapper can get a bit confusing. Maybe we could have a resource loader that loaded CPU textures from disk, and the GPU texture could have a `copy_from(CpuTexture &)` function?

Reply:
I agree, I think it would be nicer if we had some kind of separate CPU and GPU texture class, especially since both textures are in memory on different devices, so enclosing both in one wrapper can get a bit confusing. Maybe we could have a resource loader that loaded CPU textures from disk, and the GPU texture could have a copy_from(CpuTexture &) function?

Reply:
This should be updated in thew newest code changes.